### PR TITLE
Assignment Shorthands

### PR DESCRIPTION
**Pull request by hsleonis**
#### Assignment Operators

Assigning is very common. Sometimes typing becomes time consuming for us 'Lazy programmers'.
So, we can use some tricks to help us and make our code cleaner and simpler.

Instead of using:

``` javascript
a = true;
b = false;
```

We can use:

``` javascript
a |= 1;
b &= 0;
```

This is the similiar use of

``` javascript
x += 23; // x = x + 23;
y -= 15; // y = y - 15;
z *= 10; // z = z * 10;
k /= 7; // k = k / 7;
p %= 3; // p = p % 3;
d **= 2; // d = d ** 2;
m >>= 2; // m = m >> 2;
n <<= 2; // n = n << 2;
n ++; // n = n + 1;
n --; n = n - 1;

```
#### If-else (Using ternary operator)

This is what we write on regular basis.

``` javascript
var newValue;
if(value > 10) 
  newValue = 5;
else
  newValue = 2;
```

We can user ternary operator to make it awesome:

``` javascript
var newValue = (value > 10) ? 5 : 2;
```
#### Null, Undefined, Empty Checks

``` javascript
if (variable1 !== null || variable1 !== undefined || variable1 !== '') {
     var variable2 = variable1;
}
```

Shorthand here:

``` javascript
var variable2 = variable1  || '';
```

P.S.: If variable1 is a number, then first check if it is 0.
#### Object Array Notation

Instead of using:

``` javascript
var a = new Array();
a[0] = "myString1";
a[1] = "myString2";
```

Use this:

``` javascript
var a = ["myString1", "myString2"];
```
#### Associative array

Instead of using:

``` javascript
var skillSet = new Array();
skillSet['Document language'] = 'HTML5';
skillSet['Styling language'] = 'CSS3';
```

Use this:

``` javascript
var skillSet = {
    'Document language' : 'HTML5', 
    'Styling language' : 'CSS3'
};
```
## Username

Here I am: [@hsleonis](https://github.com/hsleonis)
## Website

You can see a cool animation by me: [ThemeAxe.com](http://www.themeaxe.com)

### CODE REVIEW

**Comment by loverajoel** [comment]
@hsleonis  hey use the PR template -> https://github.com/loverajoel/jstips/blob/gh-pages/GIT_TEMPLATE.md

File: _posts/en/2016-xx-xx-assignment-shorthands.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+````
+Use this:
+````javascript
+var skillSet = {
+    'Document language' : 'HTML5', 
+    'Styling language' : 'CSS3'
+};
+````
+
+## Username
```
Comment (loverajoel) [inline _posts/en/2016-xx-xx-assignment-shorthands.md]: remove it

File: _posts/en/2016-xx-xx-assignment-shorthands.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+Here I am: [@hsleonis](https://github.com/hsleonis)]
+## Website
```
Comment (loverajoel) [inline _posts/en/2016-xx-xx-assignment-shorthands.md]: same here

File: _posts/en/2016-xx-xx-assignment-shorthands.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+tip-number: 25
+tip-username: hsleonis
+tip-username-profile: https://github.com/hsleonis
+tip-tldr: Simple shorthands to make code cleaner and simpler.
+
+categories:
+    - en
+---
+
+## Assignment Shorthands
```
Comment (loverajoel) [inline _posts/en/2016-xx-xx-assignment-shorthands.md]: remove it

Comment (hsleonis) [inline _posts/en/2016-xx-xx-assignment-shorthands.md]: Can you please give some explanation why we are removing this?

Comment (loverajoel) [inline _posts/en/2016-xx-xx-assignment-shorthands.md]: Title is added automatically to the post, so this post will have two same titles

**Comment by hsleonis** [comment]
@loverajoel please check the update. I have made the changes.

**Comment by loverajoel** [comment]
@hsleonis you have to use [this](https://github.com/loverajoel/jstips/blob/gh-pages/POST_TEMPLATE.md) template for you tip and [this](https://github.com/loverajoel/jstips/blob/gh-pages/GIT_TEMPLATE.md) for you pull request message.

BTW, I like this tip

**Comment by loverajoel** [comment]
Thanks!

**Comment by yuvarajvemula** [comment]
:+1: good idea, useful tips

**Comment by hsleonis** [comment]
Thanks :)

**Pull request merged**